workflows/speciesabundance.nf

```diff
@@ -68,19 +68,39 @@ workflow SpAnce {
 
     ch_versions = Channel.empty()
 
+    // Track processed IDs
+    def processedIDs = [] as Set
+
     // Create a new channel of metadata from a sample sheet
     // NB: `input` corresponds to `params.input` and associated sample sheet schema
     input = Channel.fromSamplesheet("input")
+        .map { meta, fastq_1, fastq_2 ->
+            // Set meta.id to meta.irida_if if sample_name is not provided in the samplesheet
+            if (!meta.id) {
+                meta.id = meta.irida_id
+            } else {
+                // Replace spaces in 'id' with underscores
+                meta.id = meta.id.replaceAll(/\s+/, '_')
+            }
+            // Ensure ID is unique by appending meta.irida_id if needed
+            while (processedIDs.contains(meta.id)) {
+                meta.id = "${meta.id}_${meta.irida_id}"
+            }
```

Comment:
I think this is a useful solution to ensure unique keys, but I wonder if all of this is better serviced by unique keys with strict patterns (the same criteria as the previous IDs) handled by the `schema_input.json` code?

Reply:
I had an idea of about a different way of creating unique keys, so as to not rely on the `meta.irida_id` to append to `meta.id`.

The idea is, rather than appending `meta.irida_id` we append the index of the sample in the samplesheet.csv, so that every sample from `{1..n}` has `_1` to `_n` appended. 

My logic is that it might be less confusing to have `_n` for every sample rather than samples with that long IRIDA-Next ID appended to them. I'd suggest doing it to every sample, so users are not confused as to why it shows up. The code I was thinking would look something like this (my use of closures is amateur at best, so bear with me here)
```JAVA
Channel.fromSamplesheet("input").toList().flatMap { it.withIndex().collect{ entry, idx ->
        entry[0].id = "${entry[0].id}_${idx}"
        return [entry]
```